ConsumerApi.Tests.Integration/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -1,119 +1,109 @@
 using Backbone.BuildingBlocks.SDK.Endpoints.Common.Types;
 using Backbone.ConsumerApi.Sdk;
-using Backbone.ConsumerApi.Sdk.Authentication;
 using Backbone.ConsumerApi.Sdk.Endpoints.Challenges.Types;
-using Backbone.ConsumerApi.Tests.Integration.Configuration;
 using Backbone.ConsumerApi.Tests.Integration.Extensions;
-using Backbone.ConsumerApi.Tests.Integration.Support;
-using Microsoft.Extensions.Options;
+using static Backbone.ConsumerApi.Tests.Integration.Support.Constants;
 
 namespace Backbone.ConsumerApi.Tests.Integration.StepDefinitions;
 
 [Binding]
-[Scope(Feature = "POST Challenge")]
-[Scope(Feature = "GET Challenge")]
 internal class ChallengesApiStepDefinitions
 {
-    private string _challengeId;
-    private ApiResponse<Challenge>? _response;
-    private Client _sdk = null!;
-    private bool _isAuthenticated;
-    private readonly HttpClient _httpClient;
-    private readonly ClientCredentials _clientCredentials;
+    private readonly ChallengesContext _challengesContext;
+    private readonly IdentitiesContext _identitiesContext;
+    private readonly ResponseContext _responseContext;
 
-    public ChallengesApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, HttpClientFactory factory)
+    public ChallengesApiStepDefinitions(ChallengesContext challengesContext, IdentitiesContext identitiesContext, ResponseContext responseContext)
     {
-        _challengeId = string.Empty;
-        _httpClient = factory.CreateClient();
-        _clientCredentials = new ClientCredentials(httpConfiguration.Value.ClientCredentials.ClientId, httpConfiguration.Value.ClientCredentials.ClientSecret);
+        _challengesContext = challengesContext;
+        _identitiesContext = identitiesContext;
+        _responseContext = responseContext;
     }
 
-    [Given("the user is authenticated")]
-    public async Task GivenTheUserIsAuthenticated()
-    {
-        _sdk = await Client.CreateForNewIdentity(_httpClient, _clientCredentials, Constants.DEVICE_PASSWORD);
-        _isAuthenticated = true;
-    }
+    private Client AnonymousClient => _identitiesContext.AnonymousClient!;
+    private Client Identity(string identityName) => _identitiesContext.Identities[identityName];
+    private string ChallengeId => _challengesContext.ChallengeId!;
+    private ApiResponse<Challenge> ChallengeResponse => _responseContext.ChallengeResponse!;
```

Comment:
It's just convenience. I don't think it's not harmful to "hide" the context from which the property originates from, and then that whole block can be packed into a region and hidden. 

It's a matter of personal preference, aesthetics and style. If you'd like, I can remove it.

Reply:
Well, as mentioned in the other comment, it leads to some inconsistency. Also, we would have to copy these properties to each of the step definition files. And if someone forgets it, it gets even more inconsistent. 
In my opinion the disadvantages overweigh the advantages. But we could discuss this with the others to get some more opinions.

Reply:
Removed everything except this one since it does help and is not inconsistent:
```
private ClientPool ClientPool => _identitiesContext.ClientPool;
```